coalib/settings/ConfigurationGathering.py

```diff
@@ -30,16 +31,59 @@ def aspectize_sections(sections):
     :param sections:  List of section that potentially contain aspects setting.
     :return:          The new sections.
     """
-    for section_name, section in sections.items():
-        section.aspects = extract_aspects_from_section(section)
-        if section.aspects is not None and len(section.get('bears')):
-            logging.warning("'aspects' and 'bears' configuration is detected "
-                            "in section '{}'. Aspect-based configuration will "
-                            'takes priority and will overwrite any '
-                            'explicitly listed bears'.format(section_name))
+    for _, section in sections.items():
+        if validate_aspect_config(section):
+            section.aspects = extract_aspects_from_section(section)
+        else:
+            section.aspects = None
     return sections
 
 
+def validate_aspect_config(section):
+    """
+    Validate if a section contain required setting to run in aspects mode.
+
+    :param section: The section that potentially contain aspect
+                    setting.
+    :return:        The validity of section.
+    """
+    aspects = section.get('aspects')
+
+    if not len(aspects):
+        return False
+
+    if not section.language:
+        logging.warning('Setting `language` is not found in section `{}`. '
+                        'Usage of aspect-based setting must include '
+                        'language information.'.format(section.name))
+        return False
+
+    if len(section.get('bears')):
+        logging.warning('`aspects` and `bears` setting is detected '
+                        'in section `{}`. Aspect-based configuration will '
+                        'takes priority and will overwrite any '
+                        'explicitly listed bears.'.format(section.name))
+    return True
+
+
+def validate_and_inject_language(sections):
```

Comment:
This can be just `_set_section_language` .  validation is assumed.  doesnt need to be part of public API